### PR TITLE
Fix footer position on the login page in mobile View

### PR DESCRIPTION
**Pull request by parasharrajat**
### Fixed Issues

$ https://github.com/Expensify/App/issues/6175
$ https://github.com/Expensify/App/issues/6202

### Tests | QA Steps
1. Open Login Page. The footer should be bottom aligned.
2. Focus on the email field, continue button and email input should be visible completely.
3. Click continue,
4. click to enter the password. Password input and continue button and go backlink should be over the keyboard.

Note: **Wide Screen layout is fixed in separate [issue](https://github.com/Expensify/App/issues/5159) and which affect web | desktop | Ipad | Tablet | landscape view and the PR for the same is https://github.com/Expensify/App/pull/6137**.

### Tested On

- [ ] Web
- [x] Mobile Web
- [ ] Desktop
- [x] iOS
- [x] Android

#### Web

image

#### Mobile Web

image

#### Desktop

image

#### iOS

Screenshot 2021-11-08 01:13:25

#### Android

Screenshot 2021-11-07 19:05:24
Screenshot 2021-11-07 19:05:06
Screenshot 2021-11-07 22:45:58

### CODE REVIEW

**Review by luacmartins — CHANGES_REQUESTED** [review]
The footer shifts up as soon as we start typing on Android. Maybe extending `signInPageStyles.js/index.ios.js` to include Android would fix this.

https://user-images.githubusercontent.com/22219519/140781404-f871453b-0087-4aa0-be71-1a9c137e1954.mov

**Comment by parasharrajat** [comment]
I will check it.

**Comment by luacmartins** [comment]
@parasharrajat let me know when this is ready for another review.

**Comment by parasharrajat** [comment]
Thanks. Just trying to find a way to trigger the ui update when keyboard opens which happens when you type on input.

**Comment by parasharrajat** [comment]
@luacmartins Fixed it. Basically, when you type the LoginPage rerender but it does not when the keyboard is shown. So I created on HOC to trigger rerender when the keyboard becomes visible.

**Comment by luacmartins** [comment]
Hey @parasharrajat! I pulled the latest changes, but I still see the footer shifting up when the keyboard is displayed on Android. Maybe I'm missing something here.

**Comment by parasharrajat** [comment]
Ok. Checking now. may be I missed something.

**Comment by parasharrajat** [comment]
@luacmartins It works on all of the three devices I tested. 
Pixel 4 XL
https://user-images.githubusercontent.com/24370807/141501705-d6dd4e5a-48c6-4a6f-8f08-b2b15028c196.mp4

**Comment by luacmartins** [comment]
I see... I dislike the layout shift, but I agree that that is a minor detail here. Approving.

**Review by luacmartins — APPROVED** [review]

**Comment by ogumen** [comment]
The accessibility issue related to the **mWeb on Android**:
1. There is no possibility to zoom the page content using pinch-to-zoom on Android devices (happens due to user-scalable attribute set to "no" in `<meta name="viewport" content="width=device-width,user-scalable=no,initial-scale=1">`, the iOS devices ignore this setting) - failure of WCAG SC 1.4.4
https://user-images.githubusercontent.com/88733897/142783792-a57a7363-3b36-4142-b146-b8864b576a11.mp4

File: src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import styles from '../../../../styles/styles';
+
+const scrollViewContentContainerStyles = styles.flex1;
```
Comment (rushatgabhane) [inline src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js:3]: @parasharrajat I'm not able to understand why flex1 was used for iOS.

File: src/pages/signin/SignInPageLayout/signInPageStyles.js/index.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import styles from '../../../../styles/styles';
+
+const scrollViewContentContainerStyles = styles.mnh100;
```
Comment (rushatgabhane) [inline src/pages/signin/SignInPageLayout/signInPageStyles.js/index.js:3]: Instead of `mnh100` like here

**Review by rushatgabhane — COMMENTED** [review]
@parasharrajat just have a question here for issue 7122

File: src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js:3]: Not remember but I'm sure there was some UI thing. I will get back to you on this.

Comment (rushatgabhane) [reply src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js:3]: Cool, take your time. Thanks!

Comment (parasharrajat) [reply src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js:3]: @rushatgabhane I looked into this one. I added this as minHeight: 100% makes the Login Page width become more the screen size and thus page will become scrollable even if we have enough space to fit the whole content.

Comment (rushatgabhane) [reply src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js:3]: welp, we made iOS use minHeight 100% in this PR #7205 
This is the behaviour that we get after making the changes. It's same as using flex1 at least on the sim 😅

https://user-images.githubusercontent.com/29673073/149640142-7dcfdd8f-87a8-46fc-9ea9-17d02fb70a0f.mov

Comment (parasharrajat) [reply src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js:3]: Clear the cache and then run it. I don't remember but it i visible sometimes.

Comment (rushatgabhane) [reply src/pages/signin/SignInPageLayout/signInPageStyles.js/index.ios.js:3]: Thanks for your help. Yep, it's a problem only sometimes so missed it during review.

It's a regression. I'll comment in original issue.

**Pull request merged**